### PR TITLE
make resume available in agents view

### PR DESCRIPTION
**Pull request by kevinjosethomas**
- /resume now opens the saved-session picker from agents view before a chat exists.
- selected sessions reuse the existing agents view open flow, including active-session reuse.
- added regression coverage for command classification and saved-session resume resolution.

---

> [!NOTE]
> **Medium Risk**
> Touches session open/resume and daemon-mediated delete/rename paths; mistakes could open the wrong session or mishandle active vs on-disk sessions, though behavior is largely reused from in-session flows with new regression tests.
> 
> **Overview**
> **`/resume` is now available from the session-less agents view**, where it was previously treated as session-only.
> 
> Running it opens the same **saved-session picker overlay** (`SessionSelectorComponent`) used in interactive mode. Picking a session exits the overlay and uses the existing **open-agent** flow: live daemon sessions are matched by session file path and reused; on-disk-only sessions get an inactive `SessionSummary` via `summaryForInactiveSession`. The picker supports **rename** (daemon `rename_saved_session` when active, otherwise `SessionManager` on disk) and **delete** (daemon `delete_saved_session`).
> 
> **`SessionSelectorComponent`** now surfaces **rename/delete failures** from injected handlers and still shows success after delete/rename when a post-mutation **refresh** fails. Status auto-hide timers call **`unref`** so they do not keep the process alive.
> 
> Tests cover command whitelisting, resume summary resolution, and selector error/refresh edge cases.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 313a91664e2697793aa998324f667d5f0a949b05. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

> [!NOTE]
> ### Add resume command to agents view with session selector for saved sessions
> - Adds `resume` to the agents view command whitelist so `classifyAgentsViewCommand` routes it to the agents view handler.
> - Implements `AgentsViewMode.showSessionSelector` which renders a full-screen `SessionSelectorComponent` overlay listing current and saved sessions, resolved via `SessionManager.list` and `SessionManager.listAll`.
> - Selecting a saved session resolves to either an existing active summary (matched by session file path) or a synthesized inactive summary from saved session metadata.
> - Rename and delete operations are wired through the daemon for active sessions and directly via `SessionManager` for inactive ones; responses are strictly validated.
> - Delete and rename errors in `SessionSelectorComponent` are now caught and surfaced as status messages, with distinct messages when a subsequent session refresh fails.
>
> 
>
> <sup><a href="https://app.macroscope.com">Macroscope</a> summarized 313a916.</sup>
> 
>

### CODE REVIEW

File: packages/coding-agent/src/modes/agents-view/agents-view-mode.ts [hunk 1]
```diff
@@ -984,6 +1010,76 @@ class AgentsViewMode implements Component, Focusable {
...
+				async (onProgress) =>
+					rememberSessions(
+						await SessionManager.list(this.getSavedSessionCwd(), this.options.config.sessionDir, onProgress),
+					),
+				async (onProgress) =>
+					rememberSessions(await SessionManager.listAll(onProgress, this.options.config.sessionDir)),
+				(sessionPath) => {
+					const summary = resolveAgentsViewResumeSummary(
+						sessionPath,
+						[...savedSessionsByPath.values()],
```
[the remark this answers is not in the document]
Comment (kevinjosethomas) [reply packages/coding-agent/src/modes/agents-view/agents-view-mode.ts:1063]: Fixed in 89291bac by keeping the full daemon session list for resume resolution while preserving the filtered list for rendering.

File: packages/coding-agent/src/modes/agents-view/agents-view-mode.ts [hunk 2]
```diff
@@ -984,6 +1011,76 @@ class AgentsViewMode implements Component, Focusable {
...
+					this.finish({ type: "exit" });
+				},
+				() => this.ui.requestRender(),
+				{
+					renameSession: async (sessionPath, nextName) => {
+						const name = (nextName ?? "").trim();
+						if (!name) {
+							return;
+						}
+						SessionManager.open(sessionPath).appendSessionInfo(name);
```
[the remark this answers is not in the document]
Comment (kevinjosethomas) [reply packages/coding-agent/src/modes/agents-view/agents-view-mode.ts]: Fixed in 3c401318 by routing selector renames for daemon-resident sessions through `rename_saved_session`; inactive saved sessions still use the local session file path.

File: packages/coding-agent/src/modes/agents-view/agents-view-mode.ts [hunk 3]
```diff
@@ -984,6 +1022,92 @@ class AgentsViewMode implements Component, Focusable {
...
+				() => this.ui.requestRender(),
+				{
+					renameSession: async (sessionPath, nextName) => {
+						const name = (nextName ?? "").trim();
+						if (!name) {
+							return;
+						}
+						await this.renameSavedSessionFromSelector(sessionPath, name);
+					},
+					deleteSession: deleteSessionFile,
```
[the remark this answers is not in the document]
Comment (kevinjosethomas) [reply packages/coding-agent/src/modes/agents-view/agents-view-mode.ts]: Fixed in 692a5814 by routing resume picker deletes through the daemon `delete_saved_session` command, including the active session id when the selected file is daemon-resident.

[the remark this answers is not in the document]
Comment (kevinjosethomas) [reply packages/coding-agent/src/modes/agents-view/agents-view-mode.ts]: Fixed in 692a5814 by routing resume picker deletes through the daemon `delete_saved_session` command, including the active session id when the selected file is daemon-resident.

File: packages/coding-agent/src/modes/agents-view/agents-view-mode.ts [hunk 4]
```diff
@@ -984,6 +1035,99 @@ class AgentsViewMode implements Component, Focusable {
...
+						}
+						await this.renameSavedSessionFromSelector(sessionPath, name);
+					},
+					deleteSession: (sessionPath) => this.deleteSavedSessionFromSelector(sessionPath),
+					showRenameHint: true,
+					keybindings: this.keybindings,
+				},
+			);
+			handle = showFullPaneOverlay(this.ui, selector, 96);
+		});
```
[the remark this answers is not in the document]
Comment (kevinjosethomas) [reply packages/coding-agent/src/modes/agents-view/agents-view-mode.ts:1093]: Fixed in 5869218f by catching selector rename failures and rendering a status error. I also caught delete handler rejections so daemon-guarded delete failures surface the same way.

File: packages/coding-agent/src/modes/interactive/components/session-selector.ts [hunk 5]
```diff
@@ -814,26 +819,33 @@ export class SessionSelectorComponent extends Container implements Focusable {
 
 		// Handle session deletion
 		this.sessionList.onDeleteSession = async (sessionPath: string) => {
```
[the remark this answers is not in the document]
Comment (kevinjosethomas) [reply packages/coding-agent/src/modes/interactive/components/session-selector.ts:821]: Fixed in 68a3ba37 by narrowing delete failure handling to the delete call and keeping a successful delete reported as successful even if the follow-up refresh has trouble.

File: packages/coding-agent/src/modes/interactive/components/session-selector.ts [hunk 6]
```diff
@@ -898,8 +920,14 @@ export class SessionSelectorComponent extends Container implements Focusable {
...
 			await this.refreshSessionsAfterMutation();
-		} finally {
 			this.exitRenameMode();
+		} catch (error) {
+			this.exitRenameMode();
+			this.header.setStatusMessage(
+				{ type: "error", message: `Failed to rename: ${formatMutationError(error)}` },
+				4000,
+			);
+			this.requestRender();
```
[the remark this answers is not in the document]
Comment (kevinjosethomas) [reply packages/coding-agent/src/modes/interactive/components/session-selector.ts:928]: Fixed in 6d41dea4 by limiting `Failed to rename` to the rename call itself and keeping refresh failures separate after a successful rename.

File: packages/coding-agent/src/modes/agents-view/agents-view-mode.ts [hunk 7]
```diff
@@ -984,6 +1035,99 @@ class AgentsViewMode implements Component, Focusable {
...
+		requireDaemonData(response);
+		await this.refreshSessions();
+	}
+
+	private async deleteSavedSessionFromSelector(sessionPath: string): Promise<DeleteSessionFileResult> {
+		const response = await this.requireClient().request(
+			createAgentsViewDeleteSavedSessionCommand(sessionPath, this.lastListedSummaries),
+		);
+		return expectDeleteSessionFileResult(requireDaemonData(response));
```
[the remark this answers is not in the document]
Comment (kevinjosethomas) [reply packages/coding-agent/src/modes/agents-view/agents-view-mode.ts:1118]: Fixed in 313a9166 by omitting cached active ids from agents-view resume delete requests. The daemon still receives the delete by path and uses its live-file guard to reject currently running sessions.
